synapse/rest/media/v1/media_storage.py

```diff
@@ -127,18 +133,26 @@ async def finish():
                     f.flush()
                     f.close()
 
+                    spam = await self.spam_checker.check_media_file_for_spam(
+                        ReadableFileWrapper(self.clock, fname), file_info
+                    )
+                    if spam:
+                        logger.info("Blocking media due to spam checker")
+                        raise SpamMediaException()
```

Comment:
It might be good to document somewhere what happens in this case. It seems the file gets deleted, I assume it doesn't end up in the database?

Reply:
I've added a comment here, hopefully that is good enough?